test/models/user_test.rb

```diff
@@ -3,7 +3,11 @@
 require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test '#name_or_email' do
+    user = User.new(email: 'foo@example.com', name: '')
+    assert_equal 'foo@example.com', user.name_or_email
+
+    user = User.new(email: 'foo@example.com', name: 'foo')
+    assert_equal 'foo', user.name_or_email
```

Comment:
```suggestion
    user = User.new(email: 'foo@example.com', name: 'ふー')
    assert_equal 'ふー', user.name_or_email
```

絶対にemailから取ってきてないよね？ってわかるようなデータの方がより良いかも